src/internal/operations/test/restore_helper_test.go

```diff
@@ -0,0 +1,245 @@
+		return operations.RestoreOperation{}, nil
+	}
+
+	rod.sw = store.NewKopiaStore(rod.kms)
+
+	connectorResource := resource.Users
+	if sel.Service == selectors.ServiceSharePoint {
+		connectorResource = resource.Sites
+	}
+
+	rod.ctrl, rod.sel = ControllerWithSelector(
+		t,
+		ctx,
+		rod.acct,
+		connectorResource,
+		sel,
+		nil,
+		rod.close)
+
+	ro := newTestRestoreOp(
+		t,
+		ctx,
+		rod,
+		backupID,
+		bus,
+		ctr,
+		opts,
+		restoreCfg)
+
+	rod.sss = streamstore.NewStreamer(
+		rod.kw,
+		rod.acct.ID(),
+		rod.sel.PathService())
+
+	return ro, rod
+}
+
+// newTestRestoreOp accepts the clients required to compose a restore operation, plus
+// any other metadata, and uses them to generate a new restore operation.  This
+// allows restore chains to utilize the same temp directory and configuration
+// details.
+func newTestRestoreOp(
+	t *testing.T,
+	ctx context.Context, //revive:disable-line:context-as-argument
+	rod *restoreOpDependencies,
+	backupID model.StableID,
+	bus events.Eventer,
+	ctr *count.Bus,
+	opts control.Options,
+	restoreCfg control.RestoreConfig,
+) operations.RestoreOperation {
+	rod.ctrl.IDNameLookup = idname.NewCache(map[string]string{rod.sel.ID(): rod.sel.Name()})
+
+	ro, err := operations.NewRestoreOperation(
+		ctx,
+		opts,
+		rod.kw,
+		rod.sw,
+		rod.ctrl,
+		rod.acct,
+		backupID,
+		rod.sel,
+		restoreCfg,
+		bus,
+		ctr)
+	if !assert.NoError(t, err, clues.ToCore(err)) {
+		rod.close(t, ctx)
+		t.FailNow()
+	}
+
+	return ro
+}
+
+func runAndCheckRestore(
+	t *testing.T,
+	ctx context.Context, //revive:disable-line:context-as-argument
+	ro *operations.RestoreOperation,
+	mb *evmock.Bus,
+	expectItemsWritten int, // can be -1 to skip the check
+	acceptNoData bool,
+) *details.Details {
+	deets, err := ro.Run(ctx)
+	require.NoError(t, err, clues.ToCore(err))
+	require.NotEmpty(t, ro.Results, "the restore had non-zero results")
+	require.NotNil(t, deets, "restore details")
+
+	expectStatus := []operations.OpStatus{operations.Completed}
+	if acceptNoData {
+		expectStatus = append(expectStatus, operations.NoData)
+	}
+
+	require.Contains(
+		t,
+		expectStatus,
+		ro.Status,
+		"restore doesn't match expectation, wanted any of %v, got %s",
+		expectStatus,
+		ro.Status)
+
+	if expectItemsWritten >= 0 {
+		require.Equal(t, ro.Results.ItemsWritten, expectItemsWritten, "count of items written")
+	}
+
+	assert.NotZero(t, ro.Results.ItemsRead, "count of items read")
+	assert.NotZero(t, ro.Results.BytesRead, "bytes read")
+	assert.Equal(t, 1, ro.Results.ResourceOwners, "count of resource owners")
+	assert.NoError(t, ro.Errors.Failure(), "incremental non-recoverable error", clues.ToCore(ro.Errors.Failure()))
+	assert.Empty(t, ro.Errors.Recovered(), "incremental recoverable/iteration errors")
+	assert.Equal(t, 1, mb.TimesCalled[events.RestoreStart], "restore-start events")
+	assert.Equal(t, 1, mb.TimesCalled[events.RestoreEnd], "restore-end events")
+
+	return deets
+}
+
+type giicbcker[T any] interface {
```

Comment:
this needs a more descriptive name

Reply:
Heh, I was hoping the initialism would sufficiently pass for its func, to avoid the long name.  `GetItemsInContainerByCollisionKeyer` here we go!